### PR TITLE
Play state

### PR DESCRIPTION
**Pull request by Abhishek-90**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@Abhishek-90 Could you please create an issue for this play?

**Comment by Abhishek-90** [comment]
@atapas, Issue Created [Issue Link](https://github.com/atapas/react-play/issues/44#issue-1206762904)

**Comment by Abhishek-90** [comment]
#44

File: src/meta/play-meta.js [hunk 1]
```diff
@@ -75,6 +76,18 @@ export const plays = [
     tags: 'JSX, State, Props',
     github: 'murtuzaalisurti',
     featured: true
```
Comment (atapas) [inline src/meta/play-meta.js:78]: @Abhishek-90 There is a merge conflict in this file. Could you please resolve?

File: src/plays/index.js [hunk 2]
```diff
@@ -6,4 +6,5 @@ export { default as BasicTree } from 'plays/org-tree/BasicTree';
 export { default as MovieContainer } from 'plays/movies/MovieContainer';
 export { default as WhyReact } from 'plays/why-react/WhyReact';
 export { default as CounterApp } from 'plays/counter/CounterApp';
+export { default as States } from 'plays/states/States';
```
Comment (atapas) [inline src/plays/index.js:9]: This file too merge conflict

File: src/plays/states/States.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import { useState } from "react";
+import { useLocation } from "react-router-dom";
```
Comment (atapas) [inline src/plays/states/States.jsx:2]: Could you please drop a Readme file like this one to explain what is the project about and anything that a beginner looking into the code can take away?

https://github.com/atapas/react-play/blob/main/src/plays/social-card/Readme.md

File: src/plays/states/States.jsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import { getPlayById } from "meta/play-meta-util";
+import PlayHeader from "common/playlists/PlayHeader";
+import "./States.css";
```
Comment (atapas) [inline src/plays/states/States.jsx:5]: usually the css file names starts with small case

File: src/plays/states/States.jsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  const location = useLocation();
+  const { id } = location.state;
+  const play = getPlayById(id);
+
+  // Your Code Start below.
+  const [name, setName] = useState("");
+  const [display, setDisplay] = useState(false);
+  const [duration, setDuration] = useState(0);
+
+  const handleSubmit = () => {
```
Comment (atapas) [inline src/plays/states/States.jsx:19]: As it is a submit type button, please do a e.preventDefault

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/plays/states/States.jsx (hunk 4 above)
Comment (Abhishek-90) [reply src/plays/states/States.jsx]: Rectified!

File: src/plays/states/States.jsx (hunk 3 above)
Comment (Abhishek-90) [reply src/plays/states/States.jsx:2]: Added States.md

File: src/plays/index.js (hunk 2 above)
Comment (Abhishek-90) [reply src/plays/index.js:9]: Conflict Resolved

File: src/meta/play-meta.js (hunk 1 above)
Comment (Abhishek-90) [reply src/meta/play-meta.js:79]: Conflict Resolved

File: src/plays/states/States.jsx (hunk 5 above)
Comment (Abhishek-90) [reply src/plays/states/States.jsx]: Done!

File: src/plays/states/States.jsx (hunk 3 above)
Comment (atapas) [reply src/plays/states/States.jsx:2]: @Abhishek-90 the file name should be Readme.md

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by atapas** [comment]
Also build fails due to wrong reference of CSS file. You can run yarn build locally. @Abhishek-90

File: src/plays/states/States.jsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+import { useLocation } from "react-router-dom";
+import { getPlayById } from "meta/play-meta-util";
+import PlayHeader from "common/playlists/PlayHeader";
+import "./states.css";
```
Comment (atapas) [inline src/plays/states/States.jsx:5]: Actual file name us still with capital case. It is failing the build @Abhishek-90

**Review by atapas — CHANGES_REQUESTED** [review]

**Review by atapas — COMMENTED** [review]
@Abhishek-90 Changes are looking good now. Could you please pull the updates from main once more and notify me for merge? Also make sure, there is no warning in the console.

**Comment by Abhishek-90** [comment]
@atapas can't pull new updates from your main as there are updates on my end as well. I guess first you need to merge then I can get the combined changes

**Comment by atapas** [comment]
> @atapas can't pull new updates from your main as there are updates on my end as well. I guess first you need to merge then I can get the combined changes

Oh no @Abhishek-90 you can still merge in.. like from the atapas:main to your branch and then resolve if any conflicts. But as the PR is not showing any conflicts, I guess it is safe to merge.

**Review by atapas — APPROVED** [review]
Looks good

**Comment by atapas** [comment]
@all-contributors please add @Abhishek-90 for Code

**Pull request merged**